universal-application-tool-0.0.1/app/repository/UserRepository.java

```diff
@@ -166,4 +171,57 @@ public TrustedIntermediaryGroup createNewTrustedIntermediaryGroup(
     tiGroup.save();
     return tiGroup;
   }
+
+  public void deleteTrustedIntermediaryGroup(long id) throws NoSuchTrustedIntermediaryGroupError {
+    Optional<TrustedIntermediaryGroup> tiGroup =
+        ebeanServer.find(TrustedIntermediaryGroup.class).setId(id).findOneOrEmpty();
+    if (tiGroup.isEmpty()) {
+      throw new NoSuchTrustedIntermediaryGroupError();
+    }
+    ebeanServer.delete(tiGroup.get());
+  }
+
+  public Optional<TrustedIntermediaryGroup> getTrustedIntermediaryGroup(long id) {
+    return ebeanServer.find(TrustedIntermediaryGroup.class).setId(id).findOneOrEmpty();
+  }
+
+  public void addTrustedIntermediaryToGroup(long id, String emailAddress)
+      throws NoSuchTrustedIntermediaryGroupError {
+    Optional<TrustedIntermediaryGroup> tiGroup = getTrustedIntermediaryGroup(id);
+    if (tiGroup.isEmpty()) {
+      throw new NoSuchTrustedIntermediaryGroupError();
+    }
+    Optional<Account> accountMaybe = lookupAccount(emailAddress);
+    Account account =
+        accountMaybe.orElseGet(
+            () -> {
+              Account a = new Account();
+              a.setEmailAddress(emailAddress);
+              a.save();
+              return a;
+            });
+    account.setMemberOfGroup(tiGroup.get());
+    account.save();
+  }
+
+  public void removeTrustedIntermediaryFromGroup(long id, long accountId)
+      throws NoSuchTrustedIntermediaryGroupError, NoSuchTrustedIntermediaryError {
+    Optional<TrustedIntermediaryGroup> tiGroup = getTrustedIntermediaryGroup(id);
+    if (tiGroup.isEmpty()) {
+      throw new NoSuchTrustedIntermediaryGroupError();
+    }
+    Optional<Account> accountMaybe =
```

Comment:
(non-blocking) there could be an overloaded `lookupAccount(long accountId)` method.